packages/ui/src/views/agentflows/index.jsx

```diff
@@ -1,233 +1,243 @@
+                        nodes.forEach((node) => {
+                            if (['Multi Agents', 'Chat Models', 'Tools', 'Document Loaders'].includes(node.data.category)) {
+                                const imageSrc = `${baseURL}/api/v1/node-icon/${node.data.name}`
+                                if (!processedImages[flow.id].includes(imageSrc)) {
+                                    processedImages[flow.id].push(imageSrc)
+                                    processedNodeTypes[flow.id].push(node.data.label)
+                                }
+                            }
+                        })
                     }
-                }
-                setImages(images)
-            } catch (e) {
-                console.error(e)
+                })
+                return { processedImages, processedNodeTypes }
             }
+
+            const myAgentflowsData = getAllAgentflowsApi.data
+            const { processedImages: myImages, processedNodeTypes: myNodeTypes } = processFlowData(myAgentflowsData)
+            setMyAgentflows(myAgentflowsData)
+
+            const marketplaceAgentflows = getMarketplaceAgentflowsApi.data
+            const answerAIFlows = marketplaceAgentflows.filter((flow) => flow.type === 'Agentflow')
+            const communityFlows = marketplaceAgentflows.filter((flow) => flow.type === 'Agent Community')
+
+            const { processedImages: answerAIImages, processedNodeTypes: answerAINodeTypes } = processFlowData(answerAIFlows)
+            const { processedImages: communityImages, processedNodeTypes: communityNodeTypes } = processFlowData(communityFlows)
+
+            setAnswerAIAgentflows(answerAIFlows)
+            setCommunityAgentflows(communityFlows)
+
+            setImages({ ...myImages, ...answerAIImages, ...communityImages })
+            setNodeTypes({ ...myNodeTypes, ...answerAINodeTypes, ...communityNodeTypes })
+
+            const allFlows = [...myAgentflowsData, ...answerAIFlows, ...communityFlows]
+            const uniqueCategories = ['All', ...new Set(allFlows.flatMap((item) => (item?.category ? item.category.split(';') : [])))]
+            setCategories(uniqueCategories)
         }
-    }, [getAllAgentflows.data])
+    }, [getAllAgentflowsApi.data, getMarketplaceAgentflowsApi.data])
+
+    const filteredMyAgentflows = useMemo(() => {
+        return myAgentflows.filter((flow) => {
+            const matchesSearch =
+                flow.name.toLowerCase().includes(search.toLowerCase()) ||
+                (flow.description && flow.description.toLowerCase().includes(search.toLowerCase()))
+            const matchesCategory = categoryFilter === 'All' || (flow.category && flow.category.includes(categoryFilter))
+            return matchesSearch && matchesCategory
+        })
+    }, [myAgentflows, search, categoryFilter])
+
+    const filteredAnswerAIAgentflows = useMemo(() => {
+        return answerAIAgentflows.filter((flow) => {
```

Comment:
logic here is so similar to the above you could probably break out some of the logic used across all 3 into a separate function.